### PR TITLE
add install instructions for Spacemacs

### PR DESCRIPTION
**Pull request by ildyria**
Spacemacs does not allow the installation procedure as defined in the README.
This aims to fill that gap.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,6 +29,35 @@ F*-mode requires Emacs 24.3 or newer, and is distributed through [MELPA](https:/
...
+    ;; wrapped in a layer. If you need some configuration for these
+    ;; packages, then consider creating a layer. You can also put the
+    ;; configuration in `dotspacemacs/user-config'.
+    dotspacemacs-additional-packages '(fstar-mode)
+    ```
+
+3. Look for `dotspacemacs/user-config` and add the following lines:
+
+    ```elisp
+    (require 'helm-bookmark)
```
Comment (cpitclaudel) [inline README.md]: I don't think this is needed, is it?

File: README.md [hunk 2]
```diff
@@ -27,6 +29,35 @@ F*-mode requires Emacs 24.3 or newer, and is distributed through [MELPA](https:/
...
+    
+    (setq auto-mode-alist (cons'("\\.fst$" . fstar-mode) auto-mode-alist))
```
Comment (cpitclaudel) [inline README.md]: This isn't needed either, I believe.

File: README.md [hunk 3]
```diff
@@ -27,6 +29,35 @@ F*-mode requires Emacs 24.3 or newer, and is distributed through [MELPA](https:/
...
+
+1. Go to your `.spacemacs`.
+2. Look for `dotspacemacs-additional-packages` and add `fstar-mode`:
+
+    ```elisp
+    ;; List of additional packages that will be installed without being
+    ;; wrapped in a layer. If you need some configuration for these
+    ;; packages, then consider creating a layer. You can also put the
+    ;; configuration in `dotspacemacs/user-config'.
+    dotspacemacs-additional-packages '(fstar-mode)
```
Comment (cpitclaudel) [inline README.md]: Is this a direct edit to a customize form?

File: README.md [hunk 4]
```diff
@@ -27,6 +29,35 @@ F*-mode requires Emacs 24.3 or newer, and is distributed through [MELPA](https:/
...
+### Spacemacs
+
+To install F*-mode on Spacemacs, the method mentionned above won't work.
+
+1. Go to your `.spacemacs`.
+2. Look for `dotspacemacs-additional-packages` and add `fstar-mode`:
+
+    ```elisp
+    ;; List of additional packages that will be installed without being
+    ;; wrapped in a layer. If you need some configuration for these
```
Comment (markulf) [inline README.md]: Someone created a layer, worth mentioning? 
https://github.com/kyoDralliam/fstar-layer.git

Comment (markulf) [reply README.md]: @kkohbrok is also using spacemacs, so he might be able to give useful advice

Comment (cpitclaudel) [reply README.md]: Yes, I mentioned @kyoDralliam's neat layer it in my comment above :)

File: README.md (hunk 2 above)
Comment (ildyria) [reply README.md]: well that was to make sure to be in fstar-mode on .fst maybe not needed but it was needed for .v -> Coq in my case so.

File: README.md (hunk 1 above)
Comment (ildyria) [reply README.md]: Oups. Remnant when I tried to set up using the previous instructions emacs, it was generating an error if that was not in require.

File: README.md (hunk 3 above)
Comment (ildyria) [reply README.md]: This is an edit that allows spacemacs to download fstar-mode from MELPA.

File: README.md (hunk 2 above)
Comment (cpitclaudel) [reply README.md]: That line is already included in fstar-mode, so no need to duplicate it here

File: README.md (hunk 3 above)
Comment (cpitclaudel) [reply README.md]: I see; should we recommend that people use the customize interface to do this, though?

File: README.md (hunk 4 above)
Comment (kkohbrok) [reply README.md]: I'm using spacemacs with @kyoDralliam's layer and it works wonderfully. I just checked my .spacemas and I believe I didn't make any significant changes to make it work.

Comment (cpitclaudel) [reply README.md]: Thanks. Can someone write a two-sentences description of how to use that layer?

**Comment by ildyria** [comment]
@cpitclaudel  I updated the pull request with the instruction how to setup the layer.

**Comment by cpitclaudel** [comment]
I think this is stale; I'd still love to add spacemacs instructions (just two sentences pointing to the layer)

**Pull request closed without merging**